### PR TITLE
docs: Fix broken link and correct Header presentation when site is deployed to folder

### PR DESCRIPTION
**Pull request by lukelooker**
- Correctly constructs link to `develop` when running within a folder
- Does not show "Hamburger" menu on home page when deployed within a folder (`/components/latest`)

### Requirements

Please check the following items are addressed in your pull request (or are not applicable)

- [ ] Includes test coverage for all changes
- [ ] Documentation updated
- [ ] i18n impacts
- [ ] a11y impacts

### CODE REVIEW

File: www/src/components/HeaderContent/HeaderContent.tsx [hunk 1]
```diff
@@ -39,14 +39,16 @@ interface HeaderProps extends ThemeEditorProps {
...
+
 export const HeaderContentLayout: FC<HeaderProps> = ({
   className,
   toggleNavigation,
   updateTheme,
   hasCustomTheme,
 }) => {
   const location = useLocation()
-  const currentPath = location.pathname
+  const showHamburger = location.pathname.split('/').length < (isDev ? 2 : 4)
```
Comment (mdodgelooker) [inline www/src/components/HeaderContent/HeaderContent.tsx:51]: I'm not sure I completely follow – the toggle menu button should _not_ show on the homepage? If so, it's not working on my local.

File: www/src/pages/index.tsx [hunk 2]
```diff
@@ -97,7 +97,7 @@ const Intro = () => {
                 Install
               </Heading>
               <Paragraph fontSize="small">
-                Visit the <Link href="/develop/">Develop</Link> page for
+                Visit the <Link to="develop">Develop</Link> page for
```
Comment (mdodgelooker) [inline www/src/pages/index.tsx:100]: 👍

**Review by mdodgelooker — CHANGES_REQUESTED** [review]

**Comment by mdodgelooker** [comment]
Also, I'm curious about the coverage check failing for this PR.

File: www/src/components/HeaderContent/HeaderContent.tsx (hunk 1 above)
Comment (lukelooker) [reply www/src/components/HeaderContent/HeaderContent.tsx]: 🤦  - I tested using `window.location.pathname` initially and that does _not_ always include a leading slash `/` but `useLocation` does. Went ahead add added a `substring` to trim that off reliably.

**Review by mdodgelooker — APPROVED** [review]
✅

**Pull request merged**